hardware_interface/src/resource_manager.cpp

```diff
@@ -1074,9 +1074,13 @@ HardwareReadWriteStatus ResourceManager::read(
     {
       if (component.read(time, period) != return_type::OK)
```

Comment:
Why not simply change this?

```suggestion
      if (component.read(time, period) == return_type::ERROR)
```